beacon_chain/state/helpers.py

```diff
@@ -1,72 +1,133 @@
+    parent_hashes = (
+        active_state.recent_block_hashes[
+            cycle_length + attestation.slot - block.slot_number:
+            cycle_length * 2 + attestation.slot - block.slot_number - len(oblique_parent_hashes)
+        ] +
+        oblique_parent_hashes
 
-def get_crosslink_shards_count(active_validators_count, config=DEFAULT_CONFIG):
-    """Returns how many shards in the `crosslink_shards` list
-    """
-    shard_count = config['shard_count']
-    crosslink_shards_count = (
-        active_validators_count // config['notaries_per_crosslink']
     )
-    if crosslink_shards_count > shard_count:
-        crosslink_shards_count = shard_count
-    return crosslink_shards_count
-
-
-def get_crosslink_shards(crystallized_state, config=DEFAULT_CONFIG):
-    """Returns a list of shards that will be crosslinking
-    """
-    shard_count = config['shard_count']
-    start_from = crystallized_state.next_shard
-    if start_from >= shard_count:
-        raise ValueError(
-            'start_from %d is larger than SHARD_COUNT %d',
-            start_from,
-            shard_count,
-        )
-
-    crosslink_shards_count = get_crosslink_shards_count(
-        crystallized_state.num_active_validators,
-        config=config,
+    return parent_hashes
+
+
+def get_attestation_indices(crystallized_state,
+                            attestation,
+                            config=DEFAULT_CONFIG):
+    last_state_recalc = crystallized_state.last_state_recalc
+    cycle_length = config['cycle_length']
+    indices_for_heights = crystallized_state.indices_for_heights
+
+    shard_position = list(filter(
+        lambda x: (
+            indices_for_heights[attestation.slot - last_state_recalc + cycle_length][x].shard_id ==
+            attestation.shard_id
+        ),
+        range(len(indices_for_heights[attestation.slot - last_state_recalc + cycle_length]))
+    ))[0]
+    attestation_indices = (
+        indices_for_heights[
+            attestation.slot - last_state_recalc + cycle_length
+        ][shard_position].committee
     )
 
-    if start_from + crosslink_shards_count <= shard_count:
-        return list(range(start_from, start_from + crosslink_shards_count))
+    return attestation_indices
+
+
+def get_new_recent_block_hashes(old_block_hashes,
+                                parent_slot,
+                                current_slot,
+                                parent_hash):
+    d = current_slot - parent_slot
+    return old_block_hashes[d:] + [parent_hash] * min(d, len(old_block_hashes))
+
+
+def get_active_validator_indices(dynasty, validators):
+    o = []
+    for index, validator in enumerate(validators):
+        if (validator.start_dynasty <= dynasty and dynasty < validator.end_dynasty):
+            o.append(index)
+    return o
+
+
+def shuffle(lst,
+            seed,
+            config=DEFAULT_CONFIG):
+    lst_count = len(lst)
+    assert lst_count <= 16777216
+    rand_max = 16777216 - 16777216 % lst_count
```

Comment:
Just noted here that this function was updated in the spec yesterday. ✨ (Thanks to Paul and V!) Mind updating it in this PR?